### PR TITLE
feat: add tools - berty client

### PR DESCRIPTION
**Pull request by gfanton**

### CODE REVIEW

**Review by aeddi — APPROVED** [review]
LGTM 👍

File: tool/berty-client/client.go [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+
+// @TODO(gfanton): contact accept all
+
+func (c *client) groupNew(ctx context.Context) error {
+	res, err := c.protocol.MultiMemberGroupCreate(ctx, &protocoltypes.MultiMemberGroupCreate_Request{})
+	if err != nil {
+		return err
+	}
+
+	gpk := base64.RawStdEncoding.EncodeToString(res.GroupPK)
```
Comment (n0izn0iz) [inline tool/berty-client/client.go:75]: we use RawURLEncoding in messenger

File: tool/berty-client/client.go [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+		Link: weburl,
+	})
+
+	return err
+}
+
+// @TODO(gfanton): contact accept all
+
+func (c *client) groupNew(ctx context.Context) error {
+	res, err := c.protocol.MultiMemberGroupCreate(ctx, &protocoltypes.MultiMemberGroupCreate_Request{})
```
Comment (n0izn0iz) [inline tool/berty-client/client.go:70]: Why not use the messenger api there ?

File: tool/berty-client/client.go [hunk 3]
```diff
@@ -0,0 +1,320 @@
...
+func (c *client) groupJoin(ctx context.Context, url string) error {
+	link, err := bertylinks.UnmarshalLink(url, nil) // FIXME: support passing an optional passphrase to decrypt the link
+	if err != nil {
+		return fmt.Errorf("unalbe to unmarshal link: %w", err)
+	}
+	if !link.IsGroup() {
+		return fmt.Errorf("provided link is not a group")
+	}
+
+	_, err = c.protocol.MultiMemberGroupJoin(ctx, &protocoltypes.MultiMemberGroupJoin_Request{
```
Comment (n0izn0iz) [inline tool/berty-client/client.go:109]: Why not use the messenger api there too?

Comment (gfanton) [reply tool/berty-client/client.go:109]: I mostly taked the code from `berty mini`, I will check messenger api

**Comment by gfanton** [comment]
Outdated

**Pull request closed without merging**